pkg/controllers/v1alpha1/dataload/implement.go

```diff
@@ -329,64 +251,21 @@ func (r *DataLoadReconcilerImplement) reconcilePendingDataLoad(ctx reconcileRequ
 
-	// 1. Check if the helm release already exists
-	releaseName := utils.GetDataLoadReleaseName(ctx.Name)
-	jobName := utils.GetDataLoadJobName(releaseName)
-	existed, err := helm.CheckRelease(releaseName, ctx.Namespace)
+	// 1. Install the helm chart if not exists and requeue
+	err := engine.LoadData(ctx, targetDataload)
 	if err != nil {
-		log.Error(err, "failed to check if release exists", "releaseName", releaseName, "namespace", ctx.Namespace)
+		log.Error(err, "engine loaddata failed")
 		return utils.RequeueIfError(err)
 	}
 
-	// 2. install the helm chart if not exists and requeue
-	if !existed {
-		log.Info("DataLoad job helm chart not installed yet, will install")
-
-		targetDataset, err := utils.GetDataset(r.Client, ctx.DataLoad.Spec.Dataset.Name, ctx.DataLoad.Spec.Dataset.Namespace)
-		if err != nil {
-			log.Error(err, "targetDataset not exists", "targetDataset", releaseName, "namespace", ctx.DataLoad.Spec.Dataset.Name)
-		}
-
-		_, boundedRuntime := utils.GetRuntimeByCategory(targetDataset.Status.Runtimes, common.AccelerateCategory)
-
-		chartName := ""
-		valueFileName := ""
-		switch boundedRuntime.Type {
-		case common.ALLUXIO_RUNTIME:
-			valueFileName, err = r.generateDataLoadValueFile(ctx.DataLoad)
-			if err != nil {
-				log.Error(err, "failed to generate dataload chart's value file")
-				return utils.RequeueIfError(err)
-			}
-			chartName = utils.GetChartsDirectory() + "/" + cdataload.DATALOAD_CHART + "/" + common.ALLUXIO_RUNTIME
-		case common.JINDO_RUNTIME:
-			valueFileName, err = r.generateJindoDataLoadValueFile(ctx.DataLoad)
-			if err != nil {
-				log.Error(err, "failed to generate dataload chart's value file")
-				return utils.RequeueIfError(err)
-			}
-			chartName = utils.GetChartsDirectory() + "/" + cdataload.DATALOAD_CHART + "/" + common.JINDO_RUNTIME
-		default:
-			log.Error(fmt.Errorf("RuntimeNotSupported"), "The runtime is not supported yet", "runtime", boundedRuntime)
-			r.Recorder.Eventf(&ctx.DataLoad,
-				v1.EventTypeNormal,
-				common.RuntimeNotReady,
-				"Bounded accelerate runtime not supported")
-		}
-		err = helm.InstallRelease(releaseName, ctx.Namespace, valueFileName, chartName)
-		if err != nil {
-			log.Error(err, "failed to install dataload chart")
-			return utils.RequeueIfError(err)
-		}
-		log.Info("DataLoad job helm chart successfullly installed", "namespace", ctx.Namespace, "releaseName", releaseName)
-		r.Recorder.Eventf(&ctx.DataLoad, v1.EventTypeNormal, common.DataLoadJobStarted, "The DataLoad job %s started", jobName)
-		return utils.RequeueAfterInterval(20 * time.Second)
-	}
-
-	// 3. Check running status of the DataLoad job
+	// 2. Check running status of the DataLoad job
+	releaseName := utils.GetDataLoadReleaseName(targetDataload.Name)
+	jobName := utils.GetDataLoadJobName(releaseName)
 	log.V(1).Info("DataLoad chart already existed, check its running status")
 	job, err := utils.GetDataLoadJob(r.Client, jobName, ctx.Namespace)
```

Comment:
how about abstracting a function GetDataloadStatus in the runtime engine?
Different runtime perhpas has different implementation methods, we cannot easily get the dataload status by job status

Reply:
thx, There are two reasons. First, there are only two kinds of runtimes in a short time, and the judgment behavior of the two runtimes is the same, in order not to cause code redundancy. Second, according to the current data preheating process, this process may be universal. Of course, it makes sense to extract it also